core/tracepoint/enable_spec.rb

```diff
@@ -57,12 +57,31 @@ def test; end
 
     it 'can accept param within a block but its initialized as nil' do
       event_name = nil
-      TracePoint.new(:line) do |tp|
-        event_name = tp.event
-      end.enable do |*args|
+      trace = TracePoint.new(:line) { |tp| event_name = tp.event }
+      trace.enable do |*args|
         event_name.should equal(:line)
         args.should == [nil]
       end
+      trace.enabled?.should be_false
+    end
+
+    it 'is enabled on calling with a block if it was already enabled' do
+      event_name = nil
+      trace = TracePoint.new(:line) { |tp|event_name = tp.event }
+      trace.enable
+      trace.enabled?.should be_true
+      trace.enable { event_name.should equal(:line) }
```

Comment:
Please test the result of `enable {}`. Also, putting a `should` inside a block is not the best idea, because maybe the block is not executed at all and your test is not run at all! The only case where it's ok is if it's your only `should` in the whole test, since `mspec` would then complain that there is no `should` in your test...

Reply:
okay, I have updated the PR.